win32ss/user/ntuser/class.c

```diff
@@ -1355,9 +1355,17 @@ IntGetAtomFromStringOrAtom(
     }
     else
     {
-        ASSERT(IS_ATOM(ClassName->Buffer));
-        *Atom = (RTL_ATOM)((ULONG_PTR)ClassName->Buffer);
-        Ret = TRUE;
+        if (ClassName->Buffer)
+        {
+            *Atom = (RTL_ATOM)((ULONG_PTR)ClassName->Buffer);
+            Ret = TRUE;
+        }
+        else
+        {
+            *Atom = 0;
+            EngSetLastError(ERROR_CLASS_DOES_NOT_EXIST);
```

Comment:
This is an error code that has to be set _**by the caller of this helper function**_
Indeed, even though the string parameter is called "class" name, the function is also used with other strings that are _**NOT**_ describing window classes.

Reply:
> the caller of this helper function

The caller of this helper function is plural...